### PR TITLE
feat(onboarding): Present Terms to users upgrading from v1

### LINKED ISSUE #21113: Show Privacy Policy and Terms modal for upgraded users
When users upgrade from V1 to V2 app they need to accept updated Privacy Policy and Terms. 

For those purposes we need to show Privacy Policy modal after upgrade. 

**Designs**

https://www.figma.com/design/o4qG1bnFyuyFOvHQVGgeFY/Onboarding-for-Mobile?node-id=13865-45279&t=YzMcbvPYvKIJ31NR-1

image

### PR DESCRIPTION
**Pull request by ilmotta**
Fixes https://github.com/status-im/status-mobile/issues/21113

### Summary

We now show the onboarding intro requesting the user to accept the Terms of Use & Privacy Policy with the new button "Explore the new Status" if the user had installed **any** version of Status older than the one from this PR and had at least one _profile_.

To make this work we had to make changes in status-go. Changes in status-go are additive, with little room for regressions. status-go PR https://github.com/status-im/status-go/pull/5766

In practice, this means:

- Users coming from Status v1 who had at least one _profile_ will see the modified onboarding intro screen and will need to accept the terms to proceed.
- Users who already installed v2 and are upgrading to this PR build (devs & QAs mostly) and who had at least one _profile_ will also see the modified intro screen and will need to accept the terms to proceed.

### Platforms

- Android
- iOS

### Areas that may be impacted

Onboarding.

### Steps to test

To help with the release, I tested in both Android and iOS and couldn't find any problem. Important: due to the release schedule, this PR is being tested and will merged directly into the mobile release branch. After that I will open a PR for mobile `develop` to backport this PR.

The criteria I used to test:

```
1. Given that user Alice had installed v1 and had one or more profiles.
2. When she installs v2 and opens it, she sees the new onboarding intro and must
   agree to the terms to enable the button "Explore the new Status".
3. After pressing the button, she can login as usual in any of her profiles.
```

```
1. Given that user Alice already upgraded from v1 and accepted the terms.
2. When she reopens the app she does not need to accept terms again and can
   immediately sign-in with any of her profiles.
```

```
1. Given that user Alice already upgraded from v1 and accepted the terms.
2. When she deletes all profiles, she sees the onboarding intro for users who
   have not upgraded, i.e. she has to agree to terms and she sees the usual two
   buttons "Create profile" and "Sync or recover profile".
```

```
1. Given that user Alice never installed Status.
2. When she installs v2, she sees the normal onboarding intro screen, where she
   has to accept the terms and she sees two buttons "Create profile" and "Sync
   or recover profile".
3. When she reopens the app, she doesn't see anymore the screen to accept terms.
```

status: ready

### CODE REVIEW

**Comment by churik** [comment]
Thank you for creating user story, it is the best format to verify!

**Comment by churik** [comment]
@ilmotta thank you for amazing work!

Found one issue - if you remove all accounts after creating one PR build - user is redirected to the empty screen (**Android only**).


https://github.com/user-attachments/assets/12dad3f0-66ae-4a0c-801c-84901028e458


Issue is not reproducible on current release.
If it not easy fix, I believe we can merge with it - I'll create a followup then.

**Review by shivekkhurana — COMMENTED** [review]
LGTM, but I don't understand native code well. I'll wait for someone with Native XP to review.

**Comment by ilmotta** [comment]
Thank you for testing already @churik!

There must be something device specific, because this redirection bug doesn't happen to me (on real Android 13 device, or in the simulator with Android 14). Unfortunately, I can't fix it. As you suggested, let's go for a follow-up.

File: src/status_im/contexts/onboarding/intro/view.cljs
Comment (cammellos) [inline src/status_im/contexts/onboarding/intro/view.cljs:23]: (I know you didn't write the function and minor neatpick)
the question mark at the end of `set-terms-accepted?` threw me a bit off, I expected it to be either a function returning a boolean, or a boolean, but then you see that is an updater, `set-terms-accepted!` seems more reasonable to me since it's a side effect, took me a while to realize that its' the usestate updater

File: src/status_im/contexts/onboarding/intro/view.cljs
Comment (cammellos) [inline src/status_im/contexts/onboarding/intro/view.cljs]: minor: this could be defined outside

File: src/status_im/contexts/onboarding/intro/view.cljs
Comment (cammellos) [inline src/status_im/contexts/onboarding/intro/view.cljs]: this one as well

**Review by cammellos — APPROVED** [review]
Looks all good, I would remove the `?` at the end of `set-accepted-terms?` as I stumbled a bit on that one, but non-blocking

File: src/status_im/contexts/onboarding/intro/view.cljs
Comment (cammellos) [reply src/status_im/contexts/onboarding/intro/view.cljs:23]: I get now the reasoning, `set` + `accepted-terms?`, we are setting the `accepted-terms?` variable, I am probably a bit slow, but I found that rather confusing

File: modules/react-native-status/nodejs/status.cpp
Comment (seanstrom) [inline modules/react-native-status/nodejs/status.cpp:783]: So the code here looks good, but I'm wondering if we have to create a binding in the CPP code. I only ask because I've noticed that there are some other OBJ-C and Kotlin bindings that don't have their own CPP bindings.

For example:
https://github.com/status-im/status-mobile/blob/1cb25cf61aa51d3f0a594b5d18cb8d26205d0048/modules/react-native-status/android/src/main/java/im/status/ethereum/module/AccountManager.kt#L317-L328

So I'm wondering if we need this code, and/or if we're missing some other code.

Comment (seanstrom) [reply modules/react-native-status/nodejs/status.cpp:783]: Does someone have more context on this?
cc @siddarthkay @smohamedjavid

**Review by seanstrom — APPROVED** [review]
I think this looks good, I just have one comment about the native code. I assume the way it's written now works, but I want to make sure we're not missing some necessary code.

File: modules/react-native-status/nodejs/status.cpp
Comment (siddarthkay) [reply modules/react-native-status/nodejs/status.cpp:783]: @seanstrom : we only need cpp bindings for stuff we use in integration tests.

File: src/status_im/contexts/onboarding/intro/view.cljs
Comment (ilmotta) [reply src/status_im/contexts/onboarding/intro/view.cljs:23]: This is an interesting perspective. I agree with you this can be confusing and it only works well if everybody understands the convention `set-*`, where `*` is always identical to the read function. I wouldn't mind either way, but it's better if it's brought up to more devs to see what they think/prefer.

**Comment by ilmotta** [comment]
FYI @seanstrom, once you get the 👍🏼 from @xAlisher and merge PR https://github.com/status-im/status-mobile/pull/21083, I'll rebase and update this one.

**Pull request merged**
